### ISSUE TITLE
ESP32 Based Replacement for WiFi Module

### ISSUE THREAD

**Issue by NorthernMan54** [vincedarley/homebridge-plugin-bwaspa#27]
@vincedarley I recently created an ESP32 based replacement for the Balboa WifI module, [esp32_balboa_spa](https://github.com/NorthernMan54/esp32_balboa_spa) so that I could control it from Homebridge and other sources.

I thought I would share this here in case others are looking for options around connecting their Balboa based Spa controller to Homebridge.

Also with your plugin and the Eva app I was able to create an automation to switch the Spa between high and low heat range on a daily basis, so I could avoid running the heater when my electricity rates where highest.  And having the time sync, was also helpful, as I was able to change the timing of the filter running to the lower cost electricity times.

PS - Your plugin works really good.

**Comment by vincedarley** [comment]
Thanks - for clarity, your ESP32 replacement is effectively like-for-like, so my homebridge plugin can talk to your esp32 replacement (which I imagine is more reliable than Balboa's!)?  I love the e-ink display capability you have built - I will have to try our your work at some point... (another on the to-do list).

I do the same daily high/low automation as you - almost only ever heat my Spa with very cheap electricity. It's my favourite hands-free automation - makes me very happy!

**Comment by NorthernMan54** [comment]
Correct, my ESP32 replacement is mostly like-for-like, it does not support the Balboa cloud, but for 'LOCAL CONNECT' it works perfectly.

In regards to reliability, I don't have access to the Balboa version, so I don't know how reliable it is to compare, but I did some tricks like cache the SPA config and other values, as I found the spa controller was a bit problematic in returning them.  Currently the Balboa app, in LOCAL CONNECT works almost every time, where previously it was a bit problematic.

In regards to the ePaper display, I'm have ordered this - [ESP32-S3 Arduino LVGL WIFI&Bluetooth Development Board 4.0 "480 * 480 Smart Display 4.0 inch LCD TFT Module Capacitive Touch](https://www.aliexpress.com/item/1005006768711780.html?spm=a2g0o.productlist.main.1.3f334327dBsc4k&algo_pvid=074223da-c1fc-4c46-b0f3-2b9d67c7b565&algo_exp_id=074223da-c1fc-4c46-b0f3-2b9d67c7b565-0&pdp_npi=4%40dis%21CAD%2147.38%2138.38%21%21%2134.49%2127.94%21%402101fb0e17272756795378541ed90d%2112000038236456740%21sea%21CA%21799266116%21X&curPageLogUid=ekJxWTSeXLtN&utparam-url=scene%3Asearch%7Cquery_from%3A)

As I think this would make a better display.  If your going to buy something, you may want to wait

**Comment by vincedarley** [comment]
Closing this issue now that we link to this project in the main README.
